src/main/java/org/fineract/messagegateway/sms/service/EmailService.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Properties;
+
+
+@Service
+public class EmailService {
+
+    private static final Logger logger = LoggerFactory.getLogger(EmailService.class);
+    @Autowired
+    private JavaMailSender mailSender;
+
+    @Value("${spring.mail.host}")
+    private String smtpHost;
+
+    @Value("${spring.mail.port}")
+    private int smtpPort;
+
+    @Value("${spring.mail.username}")
+    private String smtpUsername;
+
+    @Value("${spring.mail.password}")
+    private String smtpPassword;
+
+
+
+
+    public void sendEmail(EmailRequestDTO emailRequest, String callbackUrl) {
+        boolean flag = false;
+        String error = null;
+        try{
+        ((JavaMailSenderImpl) mailSender).setHost(smtpHost);
+        ((JavaMailSenderImpl) mailSender).setPort(smtpPort);
+        ((JavaMailSenderImpl) mailSender).setUsername(smtpUsername);
+        ((JavaMailSenderImpl) mailSender).setPassword(smtpPassword);
+        Properties mailProperties = ((JavaMailSenderImpl) mailSender).getJavaMailProperties();
+        mailProperties.put("mail.smtp.connectiontimeout", 5000);
+        mailProperties.put("mail.smtp.timeout", 5000);
+        mailProperties.put("mail.smtp.writetimeout", 5000);
+        ((JavaMailSenderImpl) mailSender).setJavaMailProperties(mailProperties);
+
+        SimpleMailMessage message = new SimpleMailMessage();
+        message.setTo(emailRequest.getTo().toArray(new String[0]));
+        message.setSubject(emailRequest.getSubject());
+        message.setText(emailRequest.getBody());
+        mailSender.send(message);
+        flag = true;
+
+    } catch (Exception e) {
+        error = e.getMessage();
+    }
+        RestTemplate restTemplate = new RestTemplate();
+
+        if(flag)
+        {
+            String requestBody = "Email sent successfully to {to}";
+            restTemplate.postForObject(callbackUrl, requestBody.replace("{to}",
+                    emailRequest.getTo().toString()), String.class);
+            logger.info("Email sent to: " + emailRequest.getTo());
+        }
+        else {
+            String requestBody = "Email could not be sent to {to} because of {error} ";
+            requestBody = requestBody.replace("{error}",error);
+            restTemplate.postForObject(callbackUrl, requestBody.replace("{to}",
+                    emailRequest.getTo().toString()), String.class);
+            logger.info("Email failed to be sent because {}", error);
+
+        }
+
+
+
+    }
+    private ClientHttpRequestFactory createTimeoutRequestFactory() {
```

Comment:
could not find a usage for this method.
Again, the rest template should be create as a bean with these settings so that we don't need t create it on the fly every time. 

As rest template is going to be deprecated, better stick with webclient / retroft. There is already one implementation added to connector common